libs/chain-signatures/contract/src/lib.rs

```diff
@@ -72,80 +81,79 @@ impl Default for VersionedMpcContract {
 }
 
 #[near(serializers=[borsh])]
-#[derive(Debug, Clone)]
-pub struct AllowedCodeHash {
-    code_hash: CodeHash,
-    added: BlockHeight,
+#[derive(Debug)]
+pub struct TeeState {
+    allowed_tee_proposals: AllowedDockerImageHashes,
+    historical_tee_proposals: Vec<DockerImageHash>,
+    votes: CodeHashesVotes,
+    tee_participant_info: IterableMap<AccountId, TeeParticipantInfo>,
 }
 
-#[near(serializers=[borsh])]
-#[derive(Debug, Default)]
-pub struct AllowedCodeHashes {
-    allowed_code_hashes: Vec<AllowedCodeHash>, // ordered by `start`
+impl Default for TeeState {
+    fn default() -> Self {
+        Self {
+            allowed_tee_proposals: Default::default(),
+            historical_tee_proposals: Default::default(),
+            votes: Default::default(),
+            tee_participant_info: IterableMap::new(StorageKey::TeeParticipantInfo),
+        }
+    }
 }
 
-impl AllowedCodeHashes {
-    /// Removes all expired code hashes and returns the number of removed entries.
-    fn clean(&mut self, current_block_height: BlockHeight) -> usize {
-        // Find the first non-expired entry
-        let expired_count = self
-            .allowed_code_hashes
-            .iter()
-            .position(|entry| entry.added + TEE_UPGRADE_PERIOD >= current_block_height)
-            .unwrap_or(self.allowed_code_hashes.len());
-
-        // Remove all expired entries
-        self.allowed_code_hashes.drain(0..expired_count);
-
-        // Return the number of removed entries
-        expired_count
+impl TeeState {
+    pub fn tee_status(&self, participants: Vec<AccountId>) -> BTreeMap<AccountId, TeeQuoteStatus> {
```

Comment:
nit: comment would be good
```suggestion
    /// maps every element in `participants` to its `TeeQuoteStatus`. If an element of `participants` does not have any TEE information associated to it, then it is mapped to `TeeQuoteStatus::None`.
    pub fn tee_status(&self, participants: Vec<AccountId>) -> BTreeMap<AccountId, TeeQuoteStatus> {
```